### PR TITLE
User Viewmodel Implementation

### PR DESCRIPTION
**Pull request by Ruthy4**
setup viewmodel
view model tests

### CODE REVIEW

File: app/src/main/java/com/example/randomusergenerator/user/viewmodel/UserViewModel.kt [hunk 1]
```diff
@@ -2,33 +2,35 @@ package com.example.randomusergenerator.user.viewmodel
...
 
 @HiltViewModel
 class UserViewModel @Inject constructor(private val userRepository: UserRepository) : ViewModel() {
-    private var _userResponse = MutableStateFlow<Resource<List<UserData>>?>(null)
-    val userResponse: StateFlow<Resource<List<UserData>>?> get() = _userResponse
+    private var _uiState = MutableStateFlow(UserViewState())
+    val uiState: StateFlow<UserViewState> get() = _uiState
 
-    init {
-        getAllUsers(10)
```
Comment (princessdharmy) [inline app/src/main/java/com/example/randomusergenerator/user/viewmodel/UserViewModel.kt:20]: Why did you remove this function and the `init{}` block?

File: app/src/test/java/com/example/randomusergenerator/user/viewmodel/UserViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    private val numberOfUsers = 10
+
+    @Before
+    fun setup() {
+        MockitoAnnotations.openMocks(this)
+        serviceUnderTest = UserViewModel(repository)
+    }
+
+    @Test
+    fun `when get user is called then return users`() = runTest {
```
Comment (princessdharmy) [inline app/src/test/java/com/example/randomusergenerator/user/viewmodel/UserViewModelTest.kt:45]: when _`method_name`_ is called... Do the same for other test function names.

File: app/src/main/java/com/example/randomusergenerator/user/viewmodel/UserViewModel.kt (hunk 1 above)
Comment (Ruthy4) [reply app/src/main/java/com/example/randomusergenerator/user/viewmodel/UserViewModel.kt:20]: I wanted to call the functions in the test class. It makes it readable for me

Comment (princessdharmy) [reply app/src/main/java/com/example/randomusergenerator/user/viewmodel/UserViewModel.kt:20]: Oh! So, you will need to call the function in the view as well, right?

Comment (Ruthy4) [reply app/src/main/java/com/example/randomusergenerator/user/viewmodel/UserViewModel.kt:20]: yes. I will do that
